docs/api.md

```diff
@@ -53,13 +53,79 @@ In our experience, any system that is successful needs to grow and change as new
 
 What constitutes a compatible change and how to change the API are detailed by the [API change document](devel/api_changes.md).
 
+
+## API groups
+
+To make it easier to extend the Kubernetes API, we are in the process of implementing *API groups*.
+The API group is specified in a REST path and in the `apiVersion` field of a serialized object.
+
+Currently there are two API groups in use:
+
+1. the "core" group, which is at REST path `/api/v1` and is not specified as part of the `apiVersion` field, e.g.
+   `apiVersion: v1`.
+1. the "experimental" group, which is at REST path `/apis/experimental/$VERSION`, and which uses
+  `apiVersion: experimental/$VERSION`, where currently `apiVersion: experimental/v1alpha1`.
+  The name `experimental` implies that API groups themselves are currently under development.  The actual
+  stability of the version of objects within that group is determined by its [API
+  version](#api-versioning)
+
+In the future we expect that there will be more API groups, all at REST path `/apis/$API_GROUP` and
+using `apiVersion: $API_GROUP/$VERSION`.  We expect that there will be a way for third parties to
+create their own API groups, and to avoid naming collisions.
+
+Not all kubernetes clusters will support all API groups.  However, we intend that if an API server
+supports a group, it supports all the Kinds in the group, and all of them at the same version(s).
```

Comment:
I assume that this implies a dependency graph between Kubernetes applications and the API groups and versions that they depend upon? The same presumably applies between API groups themselves.  So for example  ApplicationX/VersionY depends upon APIGroupA/VersionB, which depends on APIGroupC/VersionD... etc.  So to move such an application between Kubernetes clusters requires that the entire dependency graph be compatible between the two clusters.  If I'm correct here, we need to be explicit about this, I think.